### PR TITLE
Ss

### PR DESCRIPTION
**Pull request by Vladyslav-Banul**

### CODE REVIEW

**Review by Daerniss — APPROVED** [review]
🔥

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,7 +1,74 @@
...
+  grades: number[],
+}
+export enum SortType {
+  Name='name',
+  Surname='surname',
+  Age='age',
+  Married='married',
+  AverageGrade='averagegrade',
+}
+export type SortOrder = 'asc' | 'desc';
```
Comment (aliceblunt8) [inline src/sortStudents.ts:18]:
```suggestion
export interface Student{
  name: string,
  surname: string,
  age: number,
  married: boolean,
  grades: number[],
}

export enum SortType {
  Name = 'name',
  Surname = 'surname',
  Age = 'age',
  Married = 'married',
  AverageGrade = 'grades',
}

export type SortOrder = 'asc' | 'desc';
```

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,7 +1,74 @@
...
+
+function calculateAvg(studentMarks: number[] = []):number {
```
Comment (aliceblunt8) [inline src/sortStudents.ts:20]:
```suggestion
function calcAvgGrade(studentMarks: number[] = []):number {
```

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,7 +1,74 @@
...
+  return studentMarks.reduce((sum, curr) => sum + curr, 0)
+    / studentMarks.length;
+}
+
+export function sortStudents(students:Student[],
+  sortBy:SortType,
+  order: SortOrder):Student[] {
```
Comment (aliceblunt8) [inline src/sortStudents.ts:27]:
```suggestion
export function sortStudents(
  students: Student[],
  sortBy: SortType,
  order: SortOrder,
  ): Student[] {
```

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,7 +1,74 @@
...
+  const copyStudents = students.map((student) => ({ ...student }));
```
Comment (aliceblunt8) [inline src/sortStudents.ts:28]: hmm why not just use spread operator? `[...students]`

File: src/sortStudents.ts [hunk 5]
```diff
@@ -1,7 +1,74 @@
...
+  const k = sortBy.toLocaleLowerCase();
```
Comment (aliceblunt8) [inline src/sortStudents.ts:29]: remove, it's already in lowercase
```suggestion
```

File: src/sortStudents.ts [hunk 6]
```diff
@@ -1,7 +1,74 @@
...
+
+  if (order === 'asc') {
```
Comment (aliceblunt8) [inline src/sortStudents.ts:31]: use switch case for sort types

File: src/sortStudents.ts [hunk 7]
```diff
@@ -1,7 +1,74 @@
...
+    if (k === SortType.AverageGrade) {
+      copyStudents.sort((a, b) => calculateAvg(a.grades)
+        - calculateAvg(b.grades));
+    } else if (k === SortType.Name || k === SortType.Surname) {
+      copyStudents.sort((a, b) => a[k].localeCompare(b[k]));
+    } else if (k === SortType.Age) {
+      copyStudents.sort((a, b) => a.age - b.age);
```
Comment (aliceblunt8) [inline src/sortStudents.ts:38]: you don't need to write each student's key: married, age, etc, your function accepts sortBy argument which represents a type of sort (a key in object)

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts (hunk 6 above)
Comment (aliceblunt8) [reply src/sortStudents.ts:33]: remove if else with SortOrder, use ternary with it inside switch case (you need to have only one switch case after all)

File: src/sortStudents.ts [hunk 8]
```diff
@@ -1,7 +1,90 @@
...
+  grades: number[],
+}
+export enum SortType {
+  Name='name',
+  Surname='surname',
+  Age='age',
+  Married='married',
+  AverageGrade='grades',
+}
+export type SortOrder = 'asc' | 'desc';
```
Comment (aliceblunt8) [inline src/sortStudents.ts:18]: add blank lines between types and interfaces

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts [hunk 9]
```diff
@@ -1,7 +1,76 @@
...
+          return -1;
+        })
+        : copyStudents.sort((b, a) => {
+          if (a[k] === b[k]) {
+            return 0;
+          }
+
+          if (a[k]) {
+            return 1;
+          }
```
Comment (OleksiiBlezniuk) [inline src/sortStudents.ts:71]:
```suggestion
      return copyStudents.sort((a, b) => order === 'asc'
        ? Number(a[k]) - Number(b[k])
        : Number(b[k]) - Number(a[k])
```

**Review by OleksiiBlezniuk — CHANGES_REQUESTED** [review]

**Review by OleksiiBlezniuk — APPROVED** [review]
Nice!

**Pull request closed without merging**